Web/Controller/Projects.hs

```diff
@@ -0,0 +1,37 @@
+module Web.Controller.Projects where
+
+import Web.Controller.Prelude
+import Web.View.Projects.Index
+import Web.View.Projects.New
+
+instance Controller ProjectsController where
+    action ProjectsAction = do
+        projects <- query @Project |> fetch
+
+        -- Fetch only specific projects by a pair of values.
+        -- In this case we really fetch all, but show how we pair the values we want to
+        -- query by.
+        let pairs = projects
+                |> fmap (\project -> "(" ++ show project.projectType ++ ", " ++ project.participants ++ ")")
+
+        projectsQuery :: [Project] <- sqlQuery "SELECT * FROM projects WHERE (project_type, participants) IN (VALUES ?)" (Only (In pairs))
```

Comment:
Here's the query. It compiles, but isn't right:


<img width="1266" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/125707/e888f475-e988-448d-82b5-363402ce3c94">

Reply:
I tried to implement ToField:

```haskell

instance ToField (ProjectType, Text) where
    toField (projectType, participants) = Many [toField projectType, toField participants]


instance Controller ProjectsController where
    action ProjectsAction = do
        projects <- query @Project |> fetch

        -- Fetch only specific projects by a pair of values.
        -- In this case we really fetch all, but show how we pair the values we want to
        -- query by.
        let pairs = projects |> fmap (\project -> (project.projectType, project.participants))

        -- Use a parameterized query to handle list of tuples
        projectsQuery :: [Project] <- sqlQuery "SELECT * FROM projects WHERE (project_type, participants) IN ?" (Only $ In pairs)


        render IndexView { .. }
```

But this now has this error

<img width="1260" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/125707/87aa4f64-fba8-44d1-b863-8891ca471fe5">